config/config.py

```
@@ -1,292 +1,340 @@
+    Returns:
+        Message indicating the result of the update
+        
+    Raises:
+        ConfigUpdateError: If the update fails
+    """
+    try:
+        # Create a temporary file in the same directory
+        config_dir = os.path.dirname(CONFIG_PATH)
+        with tempfile.NamedTemporaryFile(mode='w', dir=config_dir, delete=False) as temp_file:
+            # Load and modify configuration
+            with open(CONFIG_PATH, 'r') as config_file:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)